src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/AzureProvider.java

```diff
@@ -391,83 +401,119 @@ public void deleteInstance(Infrastructure infrastructure, String instanceId) {
                                                                                       instanceId + "'"));
 
         // Retrieve all resources attached to the instance
-        NetworkInterface networkInterface = vm.getPrimaryNetworkInterface();
-        com.microsoft.azure.management.network.Network network = networkInterface.primaryIpConfiguration().getNetwork();
-        NetworkSecurityGroup networkSecurityGroup = networkInterface.getNetworkSecurityGroup();
-        Optional<PublicIpAddress> optionalPublicIPAddress = Optional.ofNullable(vm.getPrimaryPublicIpAddress());
+        List<NetworkInterface> networkInterfaces = vm.networkInterfaceIds()
+                                                     .stream()
+                                                     .map(id -> azureService.networkInterfaces().getById(id))
+                                                     .collect(Collectors.toList());
+        List<com.microsoft.azure.management.network.Network> networks = networkInterfaces.stream()
```

Comment:
I think we can try to wrap common code here and reuse it for networkInterfaces, networks, networkSecurityGroups and publicIPAddresses. give me a shout and we do it together if you want